File: src/bench_raster.py
import time
import rasterio
from rasterio.windows import Window

FILE = "data/landsat_cog.tif"

def main():
    # We open the file (Metadata read only)
    with rasterio.open(FILE) as src:
        # Calculate center window
        c_x = src.width // 2
        c_y = src.height // 2
        w = Window(c_x, c_y, 512, 512)

        # 1. READ (Seek + Read)
        # This tests how fast the driver jumps to the byte offset
        start = time.time()
        data = src.read(1, window=w)
        elapsed = time.time() - start

        print(f"COG Window Read: {elapsed:.4f}s")
        print(f"Pixel Sum:       {data.sum()}") # Prevent dead code elimination

if __name__ == "__main__":
    main()

File: src/bench_raster.jl
using ArchGDAL
using Dates

FILE = "data/landsat_cog.tif"

ArchGDAL.read(FILE) do ds
    width = ArchGDAL.width(ds)
    height = ArchGDAL.height(ds)

    # Calculate offsets
    off_x = div(width, 2)
    off_y = div(height, 2)

    t_start = now()

    # Read band 1, at offset x,y with size 512x512
    # This maps directly to GDAL RasterIO
    data = ArchGDAL.read(ds, 1, off_x, off_y, 512, 512)

    elapsed = (now() - t_start).value / 1000.0

    println("COG Window Read: ", elapsed, "s")
    println("Pixel Sum:       ", sum(data))
end
